### PR TITLE
 scheduler(ticdc): add compat layer

### LINKED ISSUE #7720: Tracking issue for scale-out single large table to multiple TiCDC instances
This project add the ability of scale-out of a single table to multiple TiCDC nodes. Let TiCDC support dynamically split and merge spans based on statistical data, including region count, write throughput, etc. Split and merge are just table scheduling, and with the help of two-phase scheduling, they have minimal impact on replication latency.

## Span Replication

- [x] https://github.com/pingcap/tiflow/pull/7722
- [x] https://github.com/pingcap/tiflow/pull/7723
- [x] https://github.com/pingcap/tiflow/pull/7731
- [x] https://github.com/pingcap/tiflow/pull/7793
- [x] https://github.com/pingcap/tiflow/pull/7806
- [x] https://github.com/pingcap/tiflow/pull/7832
- [x] https://github.com/pingcap/tiflow/pull/7833
- [x] https://github.com/pingcap/tiflow/pull/7838
- [x] https://github.com/pingcap/tiflow/pull/7880
- [x] https://github.com/pingcap/tiflow/pull/7909
- [x] https://github.com/pingcap/tiflow/pull/7916
- [x] https://github.com/pingcap/tiflow/pull/7978
- [x] https://github.com/pingcap/tiflow/pull/8010
- [ ] Unify variable names, rename table to tableSpan
- [x] https://github.com/pingcap/tiflow/pull/8021
- [x] https://github.com/pingcap/tiflow/pull/8070
- [x] https://github.com/pingcap/tiflow/pull/8100

### Support split span by write throughput

- [x] https://github.com/pingcap/tiflow/pull/8496
- [x] https://github.com/pingcap/tiflow/pull/8505

### Document

- [x] https://github.com/pingcap/docs-cn/pull/12693
- [x] https://github.com/pingcap/docs-cn/pull/13341

### PR DESCRIPTION
**Pull request by overvenus**
### What problem does this PR solve?

The compact layer makes span replication be compatible with table replication.

Before sending messages from span replication to table replication:

 - span agent -> table scheduler
   - sets tableID base on span.TableID
 - span scheduler -> table agent
   - sets tableID base on span.TableID

After receiving messages from table replication:
  
 - table scheduler -> span agent
   - Fill span based on table ID if span is empty
 - table agent -> span scheduler
   - Fill span based on table ID if span is empty

Issue Number: ref #7720

#### Tests 

 - Unit test

##### Will it cause performance regression or break compatibility?

No.

##### Do you need to update user documentation, design documentation or monitoring documentation?

No.

### Release note 

```release-note
None
```

### CODE REVIEW

File: cdc/scheduler/internal/v3/compat/compat.go [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+var SpanReplicationMinVersion = semver.New("6.6.0-alpha")
+
+// CheckSpanReplicationEnabled check if the changefeed can enable span replication.
+func (c *Compat) CheckSpanReplicationEnabled() bool {
+	if c.hasChecked {
+		return c.enableSpanReplication
+	}
+	c.hasChecked = true
+
+	c.enableSpanReplication = c.config.RegionPerSpan != 0
```
Comment (3AceShowHand) [inline cdc/scheduler/internal/v3/compat/compat.go:79]: `config` is only used to here, so no need to keep it as a field of the `Compat`, replace it by `enableSpanReplication` when new the compat.

**Review by 3AceShowHand — APPROVED** [review]

**Review by sdojjy — APPROVED** [review]

**Comment by sdojjy** [comment]
/run-integration-test tidb=f5487e3f2596cdd63ae9840606c7fff3fb539e03

**Pull request merged**